scripts/gen_lazy_tensor.py

```diff
@@ -61,11 +61,118 @@ def node_base_ctor_call(self, schema: LazyIrSchema) -> str:
     scalar_hashes = ", ".join([f"{a.name}" for a in scalar_args])
 
     return f"""{self.node_base}(torch::lazy::OpKind({aten_symbol(schema)}),
-              {{{base_ctor_value_args}}}, std::move(shapes),
+              {{{base_ctor_value_args}}},
               [&]() {{ return {schema.node_name}OutputShape({shape_fn_inputs}); }},
               /* num_outputs */ {len(schema.returns)},
               torch::lazy::MHash({scalar_hashes}))"""
 
+  # Most of this code is copied and pasted from the upstream function at torchgen/dest/lazy_ir.py.
+  # We only remove `torch::lazy::Shape` related logic as we don't need it.
+  def gen(self, schema: LazyIrSchema) -> List[str]:
+    opkind = schema.opkind or aten_symbol(schema)
+
+    # for now, we just want one IR class decl and soon after also the method defs
+    # and we use the functional version not out/inplace.
+    all_args = schema.filtered_args()
+    value_args = schema.filtered_args(values=True, scalars=False)
+    scalar_args = schema.filtered_args(values=False, scalars=True)
+
+    ctor_args = [f"const {i.lazy_type.cpp_type()}& {i.name}" for i in all_args]
```

Comment:
I guess you can provide a method in the base class probably called: `def should_use_lazy_shape(self, schema: LazyIrSchema) -> bool:` to replace this line. The default implementation should include the exact same logic as the line specified below, and then you can override it to false. Then you don't need to copy & paste the whole `gen` here.
https://github.com/pytorch/pytorch/blob/cda0d5a57b9126c6d244fdd5b02198f05c742615/torchgen/dest/lazy_ir.py#L255

Reply:
Sounds good, I'll can update that. There is an ongoing related discussion regarding upstream codegen changes at https://github.com/pytorch/pytorch/pull/87823. I'll wait for that discussion to conclude and update this PR accordingly.

Reply:
Updated upstream lazy codegen to have boolean flag